src/binary_heap.rs

```diff
@@ -420,7 +437,25 @@ where
     }
 
     /// Pushes an item onto the binary heap without first checking if it's full.
-    #[allow(clippy::missing_safety_doc)] // TODO
+    ///
+    /// # Safety
+    ///
+    /// The binary heap must not be full.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use heapless::binary_heap::{BinaryHeap, Max};
+    ///
+    /// let mut heap: BinaryHeap<_, Max, 8> = BinaryHeap::new();
+    ///
+    /// if !heap.is_full() {
+    ///     // SAFETY: the heap is not full
```

Comment:
Same here for the `is_full` check, sorry if that was too implicit. I'd argue with

```suggestion
    ///     // SAFETY: We just created an empty heap of size 8, so it cannot be full.
```

Reply:
Done!